indico/modules/events/timetable/legacy.py

```diff
@@ -36,6 +39,11 @@ def __init__(self, management=False):
         self.management = management
 
     def serialize_timetable(self, event, days=None, hide_weekends=False, strip_empty_days=False):
+        # pre-fetch ContributionPrincipals (to save time)
+        principal_map = {contrib.id: contrib.acl_entries for contrib in Contribution.find(
```

Comment:
`Contribution.query.with_parent(event).options(joinedload('acl_entries'))`

That way you skip deleted ones since it'll use the relationship criteria to get the contributions for the event.
Also, no need for `.all()` since you iterate over it anyway